IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/WebElement.cs

```diff
@@ -0,0 +1,253 @@
+        public bool Enabled => Convert.ToBoolean(RetryAction("enabled"));
+        public bool Displayed => Convert.ToBoolean(RetryAction("displayed"));
+        public bool Selected => Convert.ToBoolean(RetryAction("selected"));
+        public Point Location => WrappedElement.Location;
+        public Size Size => WrappedElement.Size;
+
+        public WebElement FindElement(By by)
+        {
+            if(!Initialized)
+            {
+                WrappedElement = _Driver.FindElement(by);
+            }
+
+            WebDriverWait wait = new WebDriverWait(_Driver, TimeSpan.FromSeconds(5));
+            try
+            {
+                wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(by));
+            }
+            catch (WebDriverTimeoutException)
+            {
+                return new WebElement(by, _Driver);
+            }
+
+            return new WebElement(WrappedElement.FindElement(by), by, _Driver);
+        }
+
+        public ReadOnlyCollection<WebElement> FindElements(By by)
+        {
+            if (!Initialized)
+            {
+                WrappedElement = _Driver.FindElement(by);
+            }
+
+            WebDriverWait wait = new WebDriverWait(_Driver, TimeSpan.FromSeconds(5));
+            try
+            {
+                wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(by));
+            }
+            catch(WebDriverTimeoutException)
+            {
+                return new ReadOnlyCollection<WebElement>(new List<WebElement>());
+            }
+
+            return
+                new ReadOnlyCollection<WebElement>(
+                    WrappedElement.FindElements(by)
+                        .Select(webElement => 
+                        new WebElement(WrappedElement.FindElement(by), by, _Driver))
+                        .ToList());
+        }
+
+        public void Clear()
+        {
+            RetryAction("clear");
+        }
+
+        public void SendKeys(string text)
+        {
+            RetryAction("sendkeys", text);
+        }
+
+        public void Submit()
+        {
+            RetryAction("submit");
+        }
+
+        public void Click()
+        {
+            RetryAction("click");
+        }
+
+        public void Click(int timeToRetry)
+        {
+            TimeToRetry = timeToRetry;
+            Click();
+            TimeToRetry = DefaultRetryTime;
+        }
+
+        public string GetAttribute(string attributeName)
+        {
+            return RetryAction("getattribute", attributeName);
+        }
+
+        public string GetProperty(string propertyName)
+        {
+            return RetryAction("getproperty", propertyName);
+        }
+
+        public string GetCssValue(string propertyName)
+        {
+            return RetryAction("getcss", propertyName);
+        }
+
+        private const int DefaultRetryTime = 30;
+        private int TimeToRetry { get; set; } = DefaultRetryTime;
```

Comment:
I realize from usage that this implies seconds, but it might be better to either give it a more explicit name, or perhaps just use a `TimeSpan`